File: core/src/main/java/com/views/calendar/cosmocalendar/selection/criteria/month/NextMonthCriteria.java
package com.views.calendar.cosmocalendar.selection.criteria.month;

import com.views.calendar.cosmocalendar.utils.DateUtils;

import java.util.Calendar;

public class NextMonthCriteria extends BaseMonthCriteria {

    private long currentTimeInMillis;

    public NextMonthCriteria() {
        currentTimeInMillis = System.currentTimeMillis();
    }

    @Override
    protected int getMonth() {
        Calendar calendar = DateUtils.getCalendar(currentTimeInMillis);
        calendar.add(Calendar.MONTH, 1);
        return calendar.get(Calendar.MONTH);
    }

    @Override
    protected int getYear() {
        Calendar calendar = DateUtils.getCalendar(currentTimeInMillis);
        calendar.add(Calendar.MONTH, 1);
        return calendar.get(Calendar.YEAR);
    }
}

File: core/src/main/java/com/loitp/views/calendar/cosmoCalendar/selection/criteria/month/NextMonthCriteria.kt
package com.loitp.views.calendar.cosmoCalendar.selection.criteria.month

import com.loitpcore.views.calendar.cosmoCalendar.utils.DateUtils.getCalendar
import java.util.*

/**
 * Created by Loitp on 04,August,2022
 * Galaxy One company,
 * Vietnam
 * +840766040293
 * freuss47@gmail.com
 */
class NextMonthCriteria : BaseMonthCriteria() {

    private val currentTimeInMillis: Long = System.currentTimeMillis()

    override fun getMonth(): Int {
        val calendar = getCalendar(currentTimeInMillis)
        calendar.add(Calendar.MONTH, 1)
        return calendar[Calendar.MONTH]
    }

    override fun getYear(): Int {
        val calendar = getCalendar(currentTimeInMillis)
        calendar.add(Calendar.MONTH, 1)
        return calendar[Calendar.YEAR]
    }

}
